File: pm/src/main/java/org/sqlbuilder/pm/objects/VnRoleAlias.java
package org.sqlbuilder.pm.objects;

import org.sqlbuilder.common.Resolvable;
import org.sqlbuilder.common.Utils;
import org.sqlbuilder2.ser.Pair;

public class VnRoleAlias implements Resolvable<Pair<String, String>, Pair<Integer, Integer>>
{
	public String clazz;

	public String role;

	@Override
	public String dataRow()
	{
		return String.format("%s,%s", Utils.nullableQuotedEscapedString(clazz), Utils.nullableQuotedEscapedString(role));
	}

	@Override
	public Pair<String, String> resolving()
	{
		return new Pair<>(clazz, role);
	}
}

File: pm/src/main/java/org/sqlbuilder/pm/objects/VnRoleAlias.kt
package org.sqlbuilder.pm.objects

import org.sqlbuilder.common.Resolvable
import org.sqlbuilder.common.Utils.nullableQuotedEscapedString
import org.sqlbuilder.pm.PmVnRoleResolvable
import org.sqlbuilder.pm.PmVnRoleResolved

class VnRoleAlias : Resolvable<PmVnRoleResolvable, PmVnRoleResolved> {

    var clazz: String? = null

    var role: String? = null

    override fun dataRow(): String {
        return "${nullableQuotedEscapedString(clazz)},${nullableQuotedEscapedString<String?>(role)}"
    }

    override fun resolving(): Pair<String, String> {
        return Pair(clazz!!, role!!)
    }
}
